### ISSUE TITLE
Fix `length-zero-no-unit` false positives for Sass variables

### ISSUE THREAD

**Issue by ArianeBouchardConformit** [stylelint/stylelint#8034]
### What is the problem you're trying to solve?

We left the rule enabled, and it broke some of our Sass calculations.

Consider the following calculation (simplified, as getting there in our actual codebase requires a mixin and a function that would be useless to pollute this thread with).

```scss
.my-element {
  $padding-top-base-offset: 5px;
  $padding-top-extra-offset: 0px;

  padding-top: calc($padding-top-base-offset + $padding-top-extra-offset); // Simplified in the output by Sass to 5px
  padding-left: 0px;
}
```

After autofixing, Stylelint gives the following result:

```scss
  $padding-top-base-offset: 5px;
  $padding-top-extra-offset: 0;

  padding-top: calc($padding-top-base-offset + $padding-top-extra-offset); // Error!
  padding-left: 0;
```

Sass now gives me an error like this:

![Image](https://github.com/user-attachments/assets/b84f6570-53db-4a68-b77b-367c1e8c4343)

### What solution would you like to see?

It's a bit unfortunate that Sass can't handle unitless zeros in calculations (and I've opened an issue [here](https://github.com/sass/sass/issues/3959)), but unless they decide to support them, it might be wise for Stylelint to either directly implement an exception for Sass variables or give us an option to exclude properties with a regex.

Sass has historically not been very open to my improvement suggestions on philosophical/edge case grounds, so I would be grateful if you guys could move on this.

As things stand, we sadly just cannot use this rule here in our project.

**Comment by ybiquitous** [comment]
@ArianeBouchardConformit Thanks for the report. Just to be sure, does [this demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0BbAngWhgGximAdgC4AEwAOgSSQCQAOAhlFAJYEDmGREdGARgwDOMDBABmY4UXgkArHQAeAbkrV6TVhy48sCogCcGoiVJkAGRSoKqSjZm07c6MsA2xgAFOvtan-ISLikjCkANS0dpqOOjB6hsbBRACUSiQA9GkkAMosKHTYLGIsMFAkbCREABYwJBAArkR0DSR8aNlCghUQcoo2kQ4YuGLSJBbKlAC+IAA0IEW4AHIM+Aggscv5MEiCYIKCM+AQBEXsqxRUJORreoRQglcyANo21FeCRGi4BcQYkMcsnHeDAIUAY+igVxsAF1pjYrmA6u8ICgsmhiAwFA9LiA6BB3rtBBgdntIQQprM-icAGIQfQoBhEVYAK0ERwOsDo+0Q51eIHenxwbCIWKu2AZMHeV1hFzeHy+Qt+RxORKIwNB4JFIDFRAlwpmcJxeKIBKJBM12t1pKmEyAA) fit your problem?

**Comment by jeddy3** [comment]
We can check if the [declaration is standard](https://github.com/stylelint/stylelint/blob/main/lib/utils/isStandardSyntaxDeclaration.mjs), which will exclude dollar variables:

https://github.com/stylelint/stylelint/blob/3fca68706b744b18df53a2691dbd164508cd5d32/lib/rules/length-zero-no-unit/index.mjs#L123-L142

Like we do when checking at-rules:

https://github.com/stylelint/stylelint/blob/3fca68706b744b18df53a2691dbd164508cd5d32/lib/rules/length-zero-no-unit/index.mjs#L114-L121

**Comment by ybiquitous** [comment]
I see, we can just add the standard check like this: 👍🏼 

https://github.com/stylelint/stylelint/blob/3fca68706b744b18df53a2691dbd164508cd5d32/lib/rules/property-no-unknown/index.mjs#L69-L71

I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by ArianeBouchardConformit** [comment]
> [@ArianeBouchardConformit](https://github.com/ArianeBouchardConformit) Thanks for the report. Just to be sure, does [this demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0BbAngWhgGximAdgC4AEwAOgSSQCQAOAhlFAJYEDmGREdGARgwDOMDBABmY4UXgkArHQAeAbkrV6TVhy48sCogCcGoiVJkAGRSoKqSjZm07c6MsA2xgAFOvtan-ISLikjCkANS0dpqOOjB6hsbBRACUSiQA9GkkAMosKHTYLGIsMFAkbCREABYwJBAArkR0DSR8aNlCghUQcoo2kQ4YuGLSJBbKlAC+IAA0IEW4AHIM+Aggscv5MEiCYIKCM+AQBEXsqxRUJORreoRQglcyANo21FeCRGi4BcQYkMcsnHeDAIUAY+igVxsAF1pjYrmA6u8ICgsmhiAwFA9LiA6BB3rtBBgdntIQQprM-icAGIQfQoBhEVYAK0ERwOsDo+0Q51eIHenxwbCIWKu2AZMHeV1hFzeHy+Qt+RxORKIwNB4JFIDFRAlwpmcJxeKIBKJBM12t1pKmEyAA) fit your problem?

I'm really sorry for the long delay. I completely forgot. Yes. Yes it does.



> I see, we can just add the standard check like this: 👍🏼
> 
> [stylelint/lib/rules/property-no-unknown/index.mjs](https://github.com/stylelint/stylelint/blob/3fca68706b744b18df53a2691dbd164508cd5d32/lib/rules/property-no-unknown/index.mjs#L69-L71)
> 
> Lines 69 to 71 in [3fca687](/stylelint/stylelint/commit/3fca68706b744b18df53a2691dbd164508cd5d32)
>  if (!isStandardSyntaxDeclaration(decl)) { 
>  	return; 
>  } 
> 
> I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.
> 
> There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

Thanks. I really appreciate it, since as I expected, my Sass request was closed as "not planned".

Sorry for the lazy approach of "leave an issue and disappear". At first, I wasn't sure this counted as a bug, so I definitely needed validation. And now I think I might be too late to contribute on this, as the linked PR looks like it's merged. Am I right, or is there still something I can do?
